### ISSUE TITLE
Roles in parent dirs not resolved when passing ANSIBLE_ROLES_PATH

### ISSUE THREAD

**Issue by retr0h** [willthames/ansible-lint#272]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible 2.3.1.0
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, May  4 2017, 15:45:50) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]


ansible-lint 3.4.13
```

- ansible installation method: one of source, pip
- ansible-lint installation method: pip

# Desired Behaviour

Ansible lint reporting a linting issue with my role.

# Actual Behaviour (Bug report only)

A linting error is not reported.

## Docker role

```
$  ansible-lint .
$
```

## Molecule role

```
$ ansible-lint ../../../resources/roles/molecule
[ANSIBLE0011] All tasks should be named
/Users/jodewey/git/molecule_2/test/resources/roles/molecule/tasks/main.yml:2
Task/Handler: file owner=root state=directory mode=0755 dest=/etc/molecule group=root
```

So when running ansible-lint manually on the docker and molecule roles.  They both have lint errors.  However, when letting ansible-lint resolve the role directories via ANSIBLE_ROLES_PATH it does not work.

```
$ cat molecule/default/playbook.yml
---
- hosts: all
  roles:
    - molecule
    - docker
```

No error is reported, which makes sense.  It needs to know the role path or find it from ansible.cfg.

```
$ ansible-lint -v molecule/default/playbook.yml
Examining molecule/default/playbook.yml of type playbook
```

So lets set the role path.  It is *unable* to resolve the `molecule` role which is in the role path, but a few parent dirs up.

```
$ ANSIBLE_ROLES_PATH=../:../../../resources/roles/ ansible-lint -v molecule/default/playbook.yml
Examining molecule/default/playbook.yml of type playbook
Examining /Users/jodewey/git/molecule_2/test/scenarios/driver/docker/tasks/main.yml of type tasks
[ANSIBLE0012] Commands should not change things if nothing needs doing
/Users/jodewey/git/molecule_2/test/scenarios/driver/docker/tasks/main.yml:2
Task/Handler: This should pass
```

## Reproduce

```
$ git clone https://github.com/retr0h/molecule.git
$ git checkout ansible-lint-no-workie
$ cd test/scenarios/driver/docker
```

Can perform the work noted above from this directory.

**Comment by retr0h** [comment]
ping

**Comment by willthames** [comment]
Interesting:

```
> /home/will/src/ansible-lint/lib/ansiblelint/utils.py(284)_rolepath()
-> search_locations = constants.DEFAULT_ROLES_PATH
(Pdb) n
> /home/will/src/ansible-lint/lib/ansiblelint/utils.py(285)_rolepath()
-> if isinstance(search_locations, six.string_types):
(Pdb) p search_locations
[u'/etc', u'/resources/roles']
```

Not really sure why the roles path has lost its '../../../' from the start but that explains why ansible-lint is going wrong - it's getting duff information from `constants`.

**Comment by willthames** [comment]
If I run the above with ansible from devel, it all works as expected. If this is fixed in ansible 2.3.2 I'll just leave it, otherwise I'll see if there is some kind of workaround.

**Comment by retr0h** [comment]
@willthames still having the same issue with 2.4.  '../../../' is getting lost as reported originally.

**Comment by vibhor995** [comment]
[ANSIBLE0011] All tasks should be named

[ANSIBLE0012] Commands should not change things if nothing needs doing

ansible-lint giving these results....whenever trying to run yml file...

**Comment by retr0h** [comment]
Meh, `ansible-review` is the new deal.   Closing as `ansible-lint` doesn't have much movement in favor of review.
